
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Workshop on Tools for Reproducible Research">
      
      
        <meta name="author" content="Jacques Dainat et al.">
      
      
        <link rel="canonical" href="https://Juke34.github.io/workshop-reproducible-research/pages/metagenomic/07-assembly/">
      
      
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>Metagenome Assembly - Metagenomics Workshop, using Galaxy</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#assembling-reads" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../../index.md" title="Metagenomics Workshop, using Galaxy" class="md-header__button md-logo" aria-label="Metagenomics Workshop, using Galaxy" data-md-component="logo">
      
  <img src="../../images/IRD.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Metagenomics Workshop, using Galaxy
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Metagenome Assembly
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/mathis-arnal/Metagenomic_Training.git" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    mathis-arnal/Metagenomic_Training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../.." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../course-information/schedule/" class="md-tabs__link">
        
  
  
    
  
  Schedule

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../00-meta_introduction/" class="md-tabs__link">
          
  
  
    
  
  Tutorials

        </a>
      </li>
    
  

    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../../index.md" title="Metagenomics Workshop, using Galaxy" class="md-nav__button md-logo" aria-label="Metagenomics Workshop, using Galaxy" data-md-component="logo">
      
  <img src="../../images/IRD.png" alt="logo">

    </a>
    Metagenomics Workshop, using Galaxy
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/mathis-arnal/Metagenomic_Training.git" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    mathis-arnal/Metagenomic_Training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../course-information/schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Tutorials
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Tutorials
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_1" >
        
          
          <label class="md-nav__link" for="__nav_3_1" id="__nav_3_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Metagenomic
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_1">
            <span class="md-nav__icon md-icon"></span>
            Metagenomic
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../00-meta_introduction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01-background-metadata/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Starting a metagenomic project
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03-assessing-read-quality/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assessing Read Quality
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04-trimming-filtering/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Trimming and Filtering
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05-taxonomic/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Taxonomy Profiling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../06-diversity/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Taxonomic Diversity
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
        
          
          <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Galaxy
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_2">
            <span class="md-nav__icon md-icon"></span>
            Galaxy
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../galaxy/00-short_intro_galaxy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Short intro to Galaxy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/mathis-arnal/Metagenomic_Training.git/edit/main/docs/pages/metagenomic/07-assembly.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4zm10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1z"/></svg>
    </a>
  
  
    
      
    
    <a href="https://github.com/mathis-arnal/Metagenomic_Training.git/raw/main/docs/pages/metagenomic/07-assembly.md" title="View source of this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.2 8.2 0 0 1-1.23-2"/></svg>
    </a>
  


  <h1>Metagenome Assembly</h1>

<h2 id="assembling-reads">Assembling reads<a class="headerlink" href="#assembling-reads" title="Permanent link">#</a></h2>
<p>The assembly process groups reads into contigs and contigs into 
scaffolds to obtain (ideally) the sequence of a whole 
chromosome. There are many programs devoted to
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2874646/">genome</a> and 
metagenome assembly, some of the main strategies they use are Greedy extension, OLC, and De Bruijn charts. Contrary to metabarcoding, shotgun metagenomics needs an assembly step, which does not mean that metabarcoding never uses an assembly step but sometimes is unnecessary.</p>
<p><a href="{{ no such element: mkdocs.structure.pages.Page object['root'] }}/fig/03-04-01.png">
  <img src="{{ no such element: mkdocs.structure.pages.Page object['root'] }}/fig/03-04-01.png" width="868" height="777" alt="Three diagrams depicting the three assembly algorithms: The Greedy extension starts with any read, extends it whit the reads that make a match to make a contig, it continues with a different read when the previous contig can not be extended anymore. The Overlap Layout consensus finds every pairwise overlap, makes a layout graph with all the overlaps and chooses consensus sequences to make the contigs. The De Bruijn Graphs divides the reads in k-mers, makes a k-mer graph that shows all the overlapping k-mers, and chooses paths from the graph to make the contigs. " />
</a></p>
<p><a href="https://github.com/ablab/spades">MetaSPAdes</a> is an NGS de novo assembler 
for assembling large and complex metagenomics data, and it is one of the 
most used and recommended. It is part of the SPAdes toolkit, which 
contains several assembly pipelines.</p>
<p>Some of the problems faced by metagenomics assembly are:<br />
<em> Differences in coverage between the genomes due to the differences in abundance in the sample.<br />
</em> The fact that different species often share conserved regions.<br />
* The presence of several strains of a single species in the community.   </p>
<p>SPAdes already deals with the non-uniform coverage problem in its algorithm, so it is helpful for the assembly of simple communities, but the <a href="https://pubmed.ncbi.nlm.nih.gov/28298430/">metaSPAdes</a> algorithm deals with the other problems as well, allowing it to assemble metagenomes from complex communities. </p>
<p>The process of (metagenomics) assembly can take a long time, and if the connection to the server drops, the process is killed, and the process needs to restart. To avoid this, we can create a screen session.</p>
<h2 id="screen-sessions">Screen sessions<a class="headerlink" href="#screen-sessions" title="Permanent link">#</a></h2>
<h3 id="starting-a-new-session">Starting a new session<a class="headerlink" href="#starting-a-new-session" title="Permanent link">#</a></h3>
<p>A ‘session’ can be considered a new window or screen: you might open a terminal to do one thing on the computer and then open a new terminal to work on another task at the command line. You can start a session and give it a descriptive name:</p>
<p class="bash"><div class="highlight"><pre><span></span><code>$ screen -S assembly
</code></pre></div></p>
<p>This steps creates a session with the name ‘assembly’.</p>
<p>As you work, this session will stay active until you close it. Even if you log out or work on something else, the jobs you start in this session will run until completion.</p>
<h3 id="detach-session-process-keeps-running-in-the-background">Detach session (process keeps running in the background)<a class="headerlink" href="#detach-session-process-keeps-running-in-the-background" title="Permanent link">#</a></h3>
<p>You can detach from a session by pressing <code>control + a</code> followed by <code>d</code> (for detach) on your keyboard. If you reconnect to your machine, you will have to reconnect to your session to see how it went.</p>
<blockquote>
<h2 id="additional-session-commands">Additional session commands<a class="headerlink" href="#additional-session-commands" title="Permanent link">#</a></h2>
<p><strong>Seeing active sessions</strong>
If you disconnect from your session or from your ssh, you will need to reconnect to an existing 
<code>screen</code> session. You can see a list of existing sessions:
<div class="highlight"><pre><span></span><code>$ screen -ls
</code></pre></div>
{: .bash}
<strong>Reconnecting to a session</strong>
To reconnect to an existing session:</p>
<p class="bash"><div class="highlight"><pre><span></span><code>$ screen -r session_name
</code></pre></div></p>
<p>The <code>-r</code> option = 'resume  a detached screen session'</p>
<p><strong>Kill a session</strong>
To end a session, type <code>exit</code> after reconnecting to the session:</p>
<p class="callout"><div class="highlight"><pre><span></span><code>$ screen -r session_name
$ exit
</code></pre></div>
{: .bash}</p>
</blockquote>
<p class="bash">Let's see if our program is installed correctly:
<div class="highlight"><pre><span></span><code>$ metaspades.py
</code></pre></div></p>
<p class="output"><div class="highlight"><pre><span></span><code>SPAdes genome assembler v3.15.0 [metaSPAdes mode]

Usage: spades.py [options] -o &lt;output_dir&gt;

Basic options:
  -o &lt;output_dir&gt;             directory to store all the resulting files (required)
  --iontorrent                this flag is required for IonTorrent data
  --test                      runs SPAdes on a toy dataset
  -h, --help                  prints this usage message
  -v, --version               prints version

Input data:
  --12 &lt;filename&gt;             file with interlaced forward and reverse paired-end reads
  -1 &lt;filename&gt;               file with forward paired-end reads
  -2 &lt;filename&gt;               file with reverse paired-end reads    
</code></pre></div></p>
<blockquote>
<h2 id="activate-your-environment">Activate your environment<a class="headerlink" href="#activate-your-environment" title="Permanent link">#</a></h2>
<p class="callout">If you do not have the metagenomics environment activated, the previous command should have given you an error. 
Before you proceed, activate the environment:
<div class="highlight"><pre><span></span><code>conda activate metagenomics
</code></pre></div>
{: .language-bash}</p>
</blockquote>
<h2 id="metaspades-is-a-metagenomics-assembler">MetaSPAdes is a metagenomics assembler<a class="headerlink" href="#metaspades-is-a-metagenomics-assembler" title="Permanent link">#</a></h2>
<p>The help we just saw tells us how to run <code>metaspades.py</code>. We are going 
to use the most straightforward options, just specifying our forward paired-end 
reads with <code>-1</code> and reverse paired-end reads with <code>-2</code>, and the output 
directory where we want our results to be stored. 
 ~~~
<script type="math/tex"> cd ~/dc_workshop/data/trimmed_fastq
</script> metaspades.py -1 JC1A_R1.trim.fastq.gz -2 JC1A_R2.trim.fastq.gz -o ../../results/assembly_JC1A
<div class="highlight"><pre><span></span><code>{: .bash}

Now that it is running we should detach our screen with `control + a` `d` and wait for a few minutes while it running. And then attach the screen with 
`screen -r assembly` to see how it went.  

When the run is finished, it shows this message:
</code></pre></div>
======= SPAdes pipeline finished.</p>
<p>SPAdes log can be found here: /home/dcuser/dc_workshop/results/assembly_JC1A/spades.log</p>
<p>Thank you for using SPAdes!</p>
<p class="output"><div class="highlight"><pre><span></span><code>{: .bash}
Now we can kill the screen with `exit` and look at our results in the main screen.

Now, let&#39;s go to the output files: 
</code></pre></div>
<script type="math/tex"> cd ../../results/assembly_JC1A
</script> ls -F
<div class="highlight"><pre><span></span><code>{: .bash}
</code></pre></div>
assembly_graph_after_simplification.gfa  corrected/              K55/             scaffolds.fasta
assembly_graph.fastg                     dataset.info            misc/            scaffolds.paths
assembly_graph_with_scaffolds.gfa        first_pe_contigs.fasta  params.txt       spades.log
before_rr.fasta                          input_dataset.yaml      pipeline_state/  strain_graph.gfa
contigs.fasta                            K21/                    run_spades.sh    tmp/
contigs.paths                            K33/                    run_spades.yaml <br />
~~~</p>
<p>As we can see, MetaSPAdes gave us a lot of files. The ones with the assembly are the <code>contigs.fasta</code> and the <code>scaffolds.fasta</code>. 
Also, we found three <code>K</code> folders: <em>K21, K33, and K55</em>; this contains the individual result files for an assembly 
with k-mers equal to those numbers: 21, 33, and 55. The best-assembled results are 
the ones that are displayed outside these k-folders. The folder <code>corrected</code> hold the corrected reads 
with the SPAdes algorithm. Moreover, the file 
<code>assembly_graph_with_scaffolds.gfa</code> have the information needed to visualize 
our assembly by different means, like programs such as <a href="https://rrwick.github.io/Bandage/">Bandage</a>.</p>
<p>The contigs are just made from assembled reads, but the scaffolds are the result 
from a subsequent process in which the contigs are ordered, oriented, and connected with Ns.</p>
<p>We can recognize which sample our assembly outputs corresponds to because they are inside 
the assembly results folder: <code>assembly_JC1A/</code>. However, the files within it do not have the 
sample ID. If we need the files out of their folder, it is beneficial to rename them.</p>
<blockquote>
<h2 id="exercise-1-rename-all-files-in-a-folder-needed-in-the-next-episode">Exercise 1: Rename all files in a folder (needed in the next episode)<a class="headerlink" href="#exercise-1-rename-all-files-in-a-folder-needed-in-the-next-episode" title="Permanent link">#</a></h2>
<p>Add the prefix <code>JC1A</code> (the sample ID) separated by a <code>_</code> to the beginning of the names of all the contents in the <code>assembly_JC1A/</code> directory. Remember that many solutions are possible.</p>
<p>A) <code>$ mv * JC1A_</code>  <br />
B) <code>$ mv * JC1A_*</code>  <br />
C) <code>$ for name in *; do mv $name JC1A_; done</code>    <br />
D) <code>$ for name in *; do mv $name JC1A_$name; done</code>      </p>
<blockquote>
<h2 id="solution">Solution<a class="headerlink" href="#solution" title="Permanent link">#</a></h2>
<p>A)  No, this option is going to give you as error <code>mv: target 'JC1A_' is not a directory</code> 
 This is because <code>mv</code> has two options:<br />
<code>mv file_1 file_2</code><br />
<code>mv file_1, file_2, ..... file_n directory</code> <br />
 When a list of files is passed to <code>mv</code>, the <code>mv</code> expects the last parameters to be a directory.<br />
 Here, <code>*</code> gives you a list of all the files in the directory. The last parameter is <code>JC1A_</code> (which <code>mv</code> expects to be a directory). <br />
 B)  No. Again, every file is sent to the same file.<br />
 C)  No, every file is sent to the same file JC1A_<br />
 D)  Yes, this is one of the possible solutions.  </p>
<p class="challenge">¿Do you have another solution?
{: .bash}
{: .solution}</p>
</blockquote>
<h2 id="exercise-2-compare-two-fasta-files-from-the-assembly-output">Exercise 2: Compare two fasta files from the assembly output<a class="headerlink" href="#exercise-2-compare-two-fasta-files-from-the-assembly-output" title="Permanent link">#</a></h2>
<p>You want to know how many contigs and scaffolds result from the assembly. 
Use <code>contigs.fasta</code>  and <code>scaffolds.fasta</code> files and sort the commands to create correct code lines. <br />
Do they have the same number of lines? Why?<br />
<strong>Hint</strong>: You can use the following commands: <code>grep</code>, <code>|</code> (pipe), <code>-l</code>, <code>"&gt;"</code>, <code>wc</code>, <code>filename.fasta</code></p>
<blockquote>
<h2 id="solution_1">Solution<a class="headerlink" href="#solution_1" title="Permanent link">#</a></h2>
<p class="bash">~~~
<script type="math/tex"> grep '>' contigs.fasta | wc -l
</script> grep '&gt;' scaffolds.fasta | wc -l
~~~</p>
<p>A contig is created from reads and then a scaffold from a group of contigs, so we expect fewer lines in the <code>scaffolds.fasta</code> .</p>
<p>{: .solution}</p>
</blockquote>
<p>{: .challenge}</p>
<h2 id="quality-of-assemblies">Quality of assemblies<a class="headerlink" href="#quality-of-assemblies" title="Permanent link">#</a></h2>
<p class="callout">You can use several metrics to know the quality of your assemblies. <a href="https://quast.sourceforge.net/metaquast.html">MetaQuast</a> is a program
 that gives you these metrics for metagenome assemblies in an interactive report and text files and plots.</p>
</blockquote>







  
  






                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["announce.dismiss", "content.action.edit", "content.action.view", "content.code.annotate", "content.code.copy", "content.tooltips", "header.autohide", "navigation.footer", "navigation.indexes", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow", "toc.integrate"], "search": "../../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
    
  </body>
</html>