
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Workshop on Tools for Reproducible Research">
      
      
        <meta name="author" content="Jacques Dainat et al.">
      
      
        <link rel="canonical" href="https://Juke34.github.io/workshop-reproducible-research/pages/jupyter/jupyter-9-extra-material/">
      
      
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>Jupyter 9 extra material - Metagenomics Workshop, using Galaxy</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#running-jupyter-notebooks-on-a-cluster" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../../index.md" title="Metagenomics Workshop, using Galaxy" class="md-header__button md-logo" aria-label="Metagenomics Workshop, using Galaxy" data-md-component="logo">
      
  <img src="../../images/IRD.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Metagenomics Workshop, using Galaxy
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Jupyter 9 extra material
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/mathis-arnal/Metagenomic_Training.git" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    mathis-arnal/Metagenomic_Training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../.." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../course-information/schedule/" class="md-tabs__link">
        
  
  
    
  
  Schedule

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../metagenomic/00-meta_introduction/" class="md-tabs__link">
          
  
  
    
  
  Tutorials

        </a>
      </li>
    
  

    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../../index.md" title="Metagenomics Workshop, using Galaxy" class="md-nav__button md-logo" aria-label="Metagenomics Workshop, using Galaxy" data-md-component="logo">
      
  <img src="../../images/IRD.png" alt="logo">

    </a>
    Metagenomics Workshop, using Galaxy
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/mathis-arnal/Metagenomic_Training.git" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    mathis-arnal/Metagenomic_Training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../course-information/schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Tutorials
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Tutorials
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_1" >
        
          
          <label class="md-nav__link" for="__nav_3_1" id="__nav_3_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Metagenomic
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_1">
            <span class="md-nav__icon md-icon"></span>
            Metagenomic
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomic/00-meta_introduction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomic/01-background-metadata/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Starting a metagenomic project
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomic/03-assessing-read-quality/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assessing Read Quality
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomic/04-trimming-filtering/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Trimming and Filtering
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomic/05-taxonomic/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Taxonomy Profiling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomic/06-diversity/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Taxonomic Diversity
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
        
          
          <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Galaxy
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_2">
            <span class="md-nav__icon md-icon"></span>
            Galaxy
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../galaxy/00-short_intro_galaxy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Short intro to Galaxy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/mathis-arnal/Metagenomic_Training.git/edit/main/docs/pages/jupyter/jupyter-9-extra-material.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4zm10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1z"/></svg>
    </a>
  
  
    
      
    
    <a href="https://github.com/mathis-arnal/Metagenomic_Training.git/raw/main/docs/pages/jupyter/jupyter-9-extra-material.md" title="View source of this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.2 8.2 0 0 1-1.23-2"/></svg>
    </a>
  


  <h1>Jupyter 9 extra material</h1>

<p>Here are some useful resources if you want to read more about Jupyter in
general:</p>
<ul>
<li>The <a href="http://jupyter.org">Jupyter project site</a> contains a lot of information
  and inspiration.</li>
<li>The <a href="https://jupyter-notebook.readthedocs.io/en/stable/">Jupyter Notebook documentation</a>.</li>
<li>A <a href="http://ipywidgets.readthedocs.io/en/stable/index.html">guide</a> to using
  widgets for creating interactive notebooks.</li>
</ul>
<h2 id="running-jupyter-notebooks-on-a-cluster">Running jupyter notebooks on a cluster<a class="headerlink" href="#running-jupyter-notebooks-on-a-cluster" title="Permanent link">#</a></h2>
<ul>
<li>Login to Uppmax, making sure to use a specific login node, <em>e.g.</em> <code>rackham1</code>:</li>
</ul>
<div class="highlight"><pre><span></span><code>ssh &lt;your-user-name&gt;@rackham1.uppmax.uu.se
</code></pre></div>
<ul>
<li>Create/activate a conda environment containing <code>jupyter</code> then run:</li>
</ul>
<div class="highlight"><pre><span></span><code>jupyter notebook
</code></pre></div>
<p>When the Jupyter server starts up you should see something resembling:
<div class="highlight"><pre><span></span><code>[I 11:00:00.000 NotebookApp] Serving notebooks from local directory: &lt;path-to-your-local-dir&gt;
[I 11:00:00.000 NotebookApp] Jupyter Notebook 6.4.6 is running at:
[I 11:00:00.000 NotebookApp] http://localhost:8889/?token=357d65100058efa40a0641fce7005addcff339876c5e8000
[I 11:00:00.000 NotebookApp]  or http://127.0.0.1:8889/?token=357d65100058efa40a0641fce7005addcff339876c5e8000
[I 11:00:00.000 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
</code></pre></div></p>
<p>Now a Jupyter notebook server is running on the Uppmax end. The line that says:
<div class="highlight"><pre><span></span><code>[I 11:00:00.000 NotebookApp] http://localhost:8889/?token=357d65100058efa40a0641fce7005addcff339876c5e8000
</code></pre></div></p>
<p>contains information on the port used on the server side (8889 in this case) and
the token required to use the server (<code>357d65100058efa40a0641fce7005addcff339876c5e8000</code>).</p>
<p>Next step is to use this information to login to the server from your local
computer.  </p>
<p><strong>On your local computer</strong></p>
<p>In a terminal, run the following command to start port forwarding of
port 8080 on your local computer to the remote port on the Uppmax side. Replace
<remote-port> with the port given when you started the server on Uppmax. Also
replace <your-user-name> with your user name on Uppmax.</p>
<div class="highlight"><pre><span></span><code>ssh<span class="w"> </span>-N<span class="w"> </span>-L<span class="w"> </span>localhost:8080:localhost:&lt;remote-port&gt;<span class="w"> </span>&lt;your-user-name&gt;@rackham1.uppmax.uu.se
</code></pre></div>
<p>As long as this process is running the port forwarding is running. To disable it
simply interrupt it with <code>CTRL + C</code>.</p>
<p>Connect to the jupyter server by opening <code>localhost:8080</code> in your browser. When
prompted, paste the token you got when starting the server on Uppmax.</p>
<p>You are now (hopefully) accessing the jupyter server that's running on Uppmax,
via your local browser.</p>
<h2 id="integrating-notebooks-with-snakemake-workflows">Integrating notebooks with Snakemake workflows<a class="headerlink" href="#integrating-notebooks-with-snakemake-workflows" title="Permanent link">#</a></h2>
<p>In the <a href="jupyter-8-the-mrsa-case-study">case study</a> section of this tutorial we
created a Jupyter notebook that used output from a Snakemake workflow
and produced some summary results and plots. Wouldn't it be nice if this was
actually part of the workflow itself? To generate a HTML version of the notebook
we can use what we learned in the section about
<a href="jupyter-7-converting-notebooks">converting notebooks</a>. The command to execute the notebook
and save it in HTML format in a file <code>results/supplementary.html</code> would be:</p>
<div class="highlight"><pre><span></span><code>jupyter<span class="w"> </span>nbconvert<span class="w"> </span>--to<span class="w"> </span>HTML<span class="w"> </span>--output-dir<span class="w"> </span>results<span class="w"> </span>--output<span class="w"> </span>supplementary.html<span class="w"> </span>--execute<span class="w"> </span>supplementary_material.ipynb
</code></pre></div>
<p>This command could be used in a rule, <em>e.g.</em> <code>make_supplementary</code>, the input of
which would be <code>results/tables/counts.tsv</code>, <code>intermediate/multiqc_general_stats.txt</code>,
and <code>results/rulegraph.png</code>. See if you can work out how to implement such a
rule at the end of the <code>Snakefile</code> found in the <code>jupyter/</code> directory. You can
find an example in the code chunk below:</p>
<div class="highlight"><pre><span></span><code><span class="n">rule</span> <span class="n">make_supplementary</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">counts</span> <span class="o">=</span> <span class="s2">&quot;results/tables/counts.tsv&quot;</span><span class="p">,</span>
        <span class="n">summary</span> <span class="o">=</span> <span class="s2">&quot;results/tables/counts.tsv.summary&quot;</span><span class="p">,</span>
        <span class="n">multiqc_file</span> <span class="o">=</span> <span class="s2">&quot;intermediate/multiqc_general_stats.txt&quot;</span><span class="p">,</span>
        <span class="n">rulegraph</span> <span class="o">=</span> <span class="s2">&quot;results/rulegraph.png&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="s2">&quot;results/supplementary.html&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">base</span> <span class="o">=</span> <span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">output</span><span class="p">:</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">output</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span>
        <span class="nb">dir</span> <span class="o">=</span> <span class="k">lambda</span> <span class="n">wildcards</span><span class="p">,</span> <span class="n">output</span><span class="p">:</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">dirname</span><span class="p">(</span><span class="n">output</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
    <span class="n">shell</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        jupyter nbconvert --to HTML --output-dir {params.dir} --output {params.base} \</span>
<span class="sd">            --execute supplementary_material.ipynb</span>
<span class="sd">        &quot;&quot;&quot;</span>
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The Conda enivronment for the jupyter tutorial does not contain packages
required to run the full snakemake workflow. So if you wish to test jupyter
integration fully you should update the conda environment by running
<code>conda install snakemake-minimal fastqc sra-tools multiqc bowtie2 tbb
samtools htseq bedtools wget graphviz</code></p>
</div>
<h2 id="more-integrations">More integrations<a class="headerlink" href="#more-integrations" title="Permanent link">#</a></h2>
<p>Snakemake actually supports the execution of notebooks via the <code>notebook:</code>
rules directive. See more about Jupyter integration in the
<a href="https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#jupyter-notebook-integration">snakemake docs</a>.
In the <code>notebook:</code> directive of such a rule you specify the path to a jupyter
notebook (relative to the Snakefile) which is then executed when
the rule is run.</p>
<p>So how is this useful?</p>
<p>In the notebook itself this gives you access to a <code>snakemake</code> object
containing information about <strong>input</strong> and <strong>output</strong> files for the rule via
<code>snakemake.input</code> and <code>snakemake.output</code>. Similarly, you can access rule
<strong>wildcards</strong> with <code>snakemake.wildcards</code>, <strong>params</strong> with <code>snakemake.params</code>,
and <strong>config</strong> settings with <code>snakemake.config</code>.</p>
<p>When snakemake runs the rule with the <code>notebook:</code> directive <code>jupyter-nbconvert</code>
is used to execute the notebook. No HTML output is generated here but it is
possible to store a version of the notebook in its final processed form by
adding the following to the rule:</p>
<div class="highlight"><pre><span></span><code><span class="n">log</span><span class="p">:</span>
    <span class="n">notebook</span><span class="o">=</span><span class="s2">&quot;&lt;path&gt;/&lt;to&gt;/&lt;processed&gt;/&lt;notebook.ipynb&gt;&quot;</span>
</code></pre></div>
<p>Because you won't get the notebook in full HTML glory, this type of
integration is better suited if you want to use a notebook to generate figures
and store these in local files (<em>e.g.</em> pdf/svg/png formats).</p>
<p>We'll use the <code>supplementary_material.ipynb</code> notebook as an example! Let's say
that instead of exporting the entire notebook to HTML we want a rule that
outputs pdf versions of the barplot and heatmap figures we created.</p>
<p>Let's start by setting up the rule. For simplicity we'll use the same input as
when we edited the notebook in the first place. The output will be
<code>results/barplot.pdf</code> and <code>results/heatmap.pdf</code>. Let's also output a finalized
version of the notebook using the <code>log: notebook=</code> directive:</p>
<div class="highlight"><pre><span></span><code><span class="n">rule</span> <span class="n">make_supplementary_plots</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">counts</span> <span class="o">=</span> <span class="s2">&quot;results/tables/counts.tsv&quot;</span><span class="p">,</span>
        <span class="n">summary</span> <span class="o">=</span> <span class="s2">&quot;results/tables/counts.tsv.summary&quot;</span><span class="p">,</span>
        <span class="n">multiqc</span> <span class="o">=</span> <span class="s2">&quot;intermediate/multiqc_general_stats.txt&quot;</span><span class="p">,</span>
        <span class="n">rulegraph</span> <span class="o">=</span> <span class="s2">&quot;results/rulegraph.png&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">barplot</span> <span class="o">=</span> <span class="s2">&quot;results/barplot.pdf&quot;</span><span class="p">,</span>
        <span class="n">heatmap</span> <span class="o">=</span> <span class="s2">&quot;results/heatmap.pdf&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="n">notebook</span> <span class="o">=</span> <span class="s2">&quot;results/supplementary.ipynb&quot;</span>
</code></pre></div>
<p>The notebook will now have access to <code>snakemake.input.counts</code>,
<code>snakemake.output.barplot</code> and <code>snakemake.output.heatmap</code> when executed from
within the workflow. Let's go ahead and edit the notebook! In the cell where we
defined notebook parameters edit the code so that it looks like this:</p>
<div class="highlight"><pre><span></span><code><span class="n">counts_file</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">counts</span>
<span class="n">summary_file</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">summary</span>
<span class="n">multiqc_file</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">multiqc</span>
<span class="n">rulegraph_file</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">rulegraph</span>

<span class="n">SRR_IDs</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="o">.</span><span class="n">SRR_IDs</span>
<span class="n">GSM_IDs</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="o">.</span><span class="n">GSM_IDs</span>
<span class="n">GEO_ID</span><span class="o">=</span><span class="n">snakemake</span><span class="o">.</span><span class="n">params</span><span class="o">.</span><span class="n">GEO_ID</span>
</code></pre></div>
<p>Notice that we set the <code>SRR_IDs</code>, <code>GSM_IDs</code> and <code>GEO_ID</code> variables using
variables in <code>snakemake.params</code>? However, we haven't defined these in our rule
yet so let's go ahead and do that now. Add the <code>params</code> section so that the
<code>make_supplementary_plots</code> in the Snakefile looks like this:</p>
<div class="highlight"><pre><span></span><code><span class="n">rule</span> <span class="n">make_supplementary_plots</span><span class="p">:</span>
    <span class="nb">input</span><span class="p">:</span>
        <span class="n">counts</span> <span class="o">=</span> <span class="s2">&quot;results/tables/counts.tsv&quot;</span><span class="p">,</span>
        <span class="n">multiqc</span> <span class="o">=</span> <span class="s2">&quot;intermediate/multiqc_general_stats.txt&quot;</span><span class="p">,</span>
        <span class="n">rulegraph</span> <span class="o">=</span> <span class="s2">&quot;results/rulegraph.png&quot;</span>
    <span class="n">output</span><span class="p">:</span>
        <span class="n">barplot</span> <span class="o">=</span> <span class="s2">&quot;results/barplot.pdf&quot;</span><span class="p">,</span>
        <span class="n">heatmap</span> <span class="o">=</span> <span class="s2">&quot;results/heatmap.pdf&quot;</span>
    <span class="n">log</span><span class="p">:</span>
        <span class="n">notebook</span> <span class="o">=</span> <span class="s2">&quot;results/supplementary.ipynb&quot;</span>
    <span class="n">params</span><span class="p">:</span>
        <span class="n">SRR_IDs</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;SRR935090&quot;</span><span class="p">,</span><span class="s2">&quot;SRR935091&quot;</span><span class="p">,</span><span class="s2">&quot;SRR935092&quot;</span><span class="p">],</span>
        <span class="n">GSM_IDs</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;GSM1186459&quot;</span><span class="p">,</span> <span class="s2">&quot;GSM1186460&quot;</span><span class="p">,</span> <span class="s2">&quot;GSM1186461&quot;</span><span class="p">],</span>
        <span class="n">GEO_ID</span> <span class="o">=</span> <span class="s2">&quot;GSE48896&quot;</span>
    <span class="n">notebook</span><span class="p">:</span> <span class="s2">&quot;supplementary_material.ipynb&quot;</span>
</code></pre></div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>One way to further generalize this rule could be to define the SRR_IDs, GSM_IDs
and GEO_ID parameters in a config file instead, in which case they would be
directly accessible from within the notebook using <code>snakemake.config['SRR_IDs']</code>
etc.</p>
</div>
<p>Now the rule contains everything needed, but we still need to edit the notebook
to save the plots to the output files. First, edit the cell that generates the
barplot so that it looks like this:</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Create a stacked barplot</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">summary_plot_data</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">kind</span><span class="o">=</span><span class="s2">&quot;bar&quot;</span><span class="p">,</span> <span class="n">stacked</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">colors</span><span class="p">)</span>
<span class="c1"># Move legend and set legend title</span>
<span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Category&quot;</span><span class="p">);</span>
<span class="n">plt</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="o">.</span><span class="n">barplot</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">300</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s2">&quot;tight&quot;</span><span class="p">)</span> <span class="c1">## &lt;-- Add this line!</span>
</code></pre></div>
<p>Finally, edit the cell that generates the heatmap so that it looks like this:</p>
<div class="highlight"><pre><span></span><code><span class="n">count_data</span> <span class="o">=</span> <span class="n">counts</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">SRR_IDs</span><span class="p">]</span>
<span class="n">heatmap_data</span> <span class="o">=</span> <span class="n">count_data</span><span class="o">.</span><span class="n">loc</span><span class="p">[(</span><span class="n">count_data</span><span class="o">.</span><span class="n">std</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">div</span><span class="p">(</span><span class="n">count_data</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">))</span><span class="o">&gt;</span><span class="mf">1.2</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">count_data</span><span class="o">.</span><span class="n">max</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">&gt;</span><span class="mi">5</span><span class="p">)]</span>
<span class="n">heatmap_data</span> <span class="o">=</span> <span class="n">heatmap_data</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span> <span class="o">=</span> <span class="n">name_dict</span><span class="p">[</span><span class="s1">&#39;title&#39;</span><span class="p">])</span>
<span class="k">with</span> <span class="n">sns</span><span class="o">.</span><span class="n">plotting_context</span><span class="p">(</span><span class="s2">&quot;notebook&quot;</span><span class="p">,</span> <span class="n">font_scale</span><span class="o">=</span><span class="mf">0.7</span><span class="p">):</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">clustermap</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">log10</span><span class="p">(</span><span class="n">heatmap_data</span><span class="o">+</span><span class="mi">1</span><span class="p">),</span> <span class="n">cmap</span><span class="o">=</span><span class="s2">&quot;RdYlBu_r&quot;</span><span class="p">,</span>
                        <span class="n">method</span><span class="o">=</span><span class="s2">&quot;complete&quot;</span><span class="p">,</span> <span class="n">yticklabels</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">.5</span><span class="p">,</span>
                        <span class="n">cbar_pos</span><span class="o">=</span><span class="p">(</span><span class="mf">.7</span><span class="p">,</span> <span class="mf">.85</span><span class="p">,</span> <span class="mf">.05</span><span class="p">,</span> <span class="mf">.1</span><span class="p">),</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span><span class="mi">9</span><span class="p">))</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">setp</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">ax_heatmap</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">(),</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">270</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="n">snakemake</span><span class="o">.</span><span class="n">output</span><span class="o">.</span><span class="n">heatmap</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">300</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s2">&quot;tight&quot;</span><span class="p">)</span> <span class="c1">## &lt;-- Add this line!</span>
</code></pre></div>
<p>Now you can run the following to generate the plots:</p>
<div class="highlight"><pre><span></span><code>snakemake<span class="w"> </span>-j<span class="w"> </span><span class="m">1</span><span class="w"> </span>make_supplementary_plots
</code></pre></div>
<h2 id="presentations-with-jupyter">Presentations with Jupyter<a class="headerlink" href="#presentations-with-jupyter" title="Permanent link">#</a></h2>
<p>As if all the above wasn't enough you can also create presentations/slideshows
with Jupyter! Simply use conda to install the
<a href="https://rise.readthedocs.io/en/stable/">RISE</a> extension to your jupyter
environment:</p>
<div class="highlight"><pre><span></span><code>conda<span class="w"> </span>install<span class="w"> </span>-c<span class="w"> </span>conda-forge<span class="w"> </span>rise
</code></pre></div>
<p>then open up a notebook of your choice. In the menu click <strong>View</strong> -&gt; <strong>Cell
Toolbar</strong> -&gt; <strong>Slideshow</strong>. Now every cell will have a drop-down in the upper
right corner allowing you to set the cell type:</p>
<ul>
<li><strong>Slide</strong>: a regular slide</li>
<li><strong>Sub-Slide</strong>: a regular slide that will be displayed <em>below</em> the previous</li>
<li><strong>Fragment</strong>: these cells split up slides so that content (fragments) are
  added only when you press Space</li>
<li><strong>Skip</strong>: these cells will not appear in the presentation</li>
<li><strong>Notes</strong>: these cells act as notes, shown in the speaker view but not in the
  main view</li>
</ul>
<p>The presentation can be run directly from the notebook by clicking the
'Enter/Exit RISE Slideshow' button (looks like a bar chart) in the toolbar, or
by using the keyboard shortcut <code>Alt-r</code>. Running it directly from a notebook
means you can also edit and run cells during your presentation. The downside is
that the presentation is not as portable because it may rely on certain software
packages that other people are not comfortable with installing.</p>
<p>You can also export the notebook to an HTML-file with <code>jupyter nbconvert
--execute --to SLIDES &lt;your-notebook.ipynb&gt;</code>. The resulting file, with the
slideshow functionality included, can be opened in any browser. However, in
this format you cannot run/edit cells.</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["announce.dismiss", "content.action.edit", "content.action.view", "content.code.annotate", "content.code.copy", "content.tooltips", "header.autohide", "navigation.footer", "navigation.indexes", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow", "toc.integrate"], "search": "../../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
    
  </body>
</html>