
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Workshop on Tools for Reproducible Research">
      
      
        <meta name="author" content="Jacques Dainat et al.">
      
      
        <link rel="canonical" href="https://Juke34.github.io/workshop-reproducible-research/pages/metagenomic/04-trimming-filtering/">
      
      
        <link rel="prev" href="../03-assessing-read-quality/">
      
      
        <link rel="next" href="../05-taxonomic/">
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>Trimming and Filtering - Metagenomics Workshop, using Galaxy</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#cleaning-reads" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../../index.md" title="Metagenomics Workshop, using Galaxy" class="md-header__button md-logo" aria-label="Metagenomics Workshop, using Galaxy" data-md-component="logo">
      
  <img src="../../images/IRD.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Metagenomics Workshop, using Galaxy
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Trimming and Filtering
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/mathis-arnal/Metagenomic_Training.git" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    mathis-arnal/Metagenomic_Training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../.." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../course-information/schedule/" class="md-tabs__link">
        
  
  
    
  
  Schedule

      </a>
    </li>
  

      
        
  
  
  
    
  
  
    
    
      
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../00-meta_introduction/" class="md-tabs__link">
          
  
  
    
  
  Tutorials

        </a>
      </li>
    
  

    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../../index.md" title="Metagenomics Workshop, using Galaxy" class="md-nav__button md-logo" aria-label="Metagenomics Workshop, using Galaxy" data-md-component="logo">
      
  <img src="../../images/IRD.png" alt="logo">

    </a>
    Metagenomics Workshop, using Galaxy
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/mathis-arnal/Metagenomic_Training.git" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    mathis-arnal/Metagenomic_Training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../course-information/schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Tutorials
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Tutorials
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_1" checked>
        
          
          <label class="md-nav__link" for="__nav_3_1" id="__nav_3_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Metagenomic
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3_1">
            <span class="md-nav__icon md-icon"></span>
            Metagenomic
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../00-meta_introduction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01-background-metadata/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Starting a metagenomic project
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03-assessing-read-quality/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assessing Read Quality
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Trimming and Filtering
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Trimming and Filtering
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#cleaning-reads" class="md-nav__link">
    <span class="md-ellipsis">
      Cleaning reads
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trimmomatic-options" class="md-nav__link">
    <span class="md-ellipsis">
      Trimmomatic options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-trimmomatic-on-galaxy" class="md-nav__link">
    <span class="md-ellipsis">
      Running Trimmomatic on Galaxy
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running Trimmomatic on Galaxy">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-a-new-history" class="md-nav__link">
    <span class="md-ellipsis">
      Create a new history
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exercise-1-what-did-trimmomatic-do" class="md-nav__link">
    <span class="md-ellipsis">
      Exercise 1: What did Trimmomatic do?
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#solution" class="md-nav__link">
    <span class="md-ellipsis">
      Solution
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exercise-2" class="md-nav__link">
    <span class="md-ellipsis">
      Exercise 2
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bonus-exercise-quality-test-after-trimming" class="md-nav__link">
    <span class="md-ellipsis">
      Bonus Exercise: Quality test after trimming
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05-taxonomic/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Taxonomy Profiling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../06-diversity/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Taxonomic Diversity
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
        
          
          <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Galaxy
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_2">
            <span class="md-nav__icon md-icon"></span>
            Galaxy
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../galaxy/00-short_intro_galaxy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Short intro to Galaxy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/mathis-arnal/Metagenomic_Training.git/edit/main/docs/pages/metagenomic/04-trimming-filtering.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4zm10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1z"/></svg>
    </a>
  
  
    
      
    
    <a href="https://github.com/mathis-arnal/Metagenomic_Training.git/raw/main/docs/pages/metagenomic/04-trimming-filtering.md" title="View source of this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.2 8.2 0 0 1-1.23-2"/></svg>
    </a>
  


  <h1>Trimming and Filtering</h1>

<h2 id="cleaning-reads">Cleaning reads<a class="headerlink" href="#cleaning-reads" title="Permanent link">#</a></h2>
<p>In the last episode, we took a high-level look at the quality
of each of our samples using <code>FastQC</code>. We visualized per-base quality
graphs showing the distribution of the quality at each base across
all the reads from our sample. This information helps us to determine 
the quality threshold we will accept, and thus, we saw information about
which samples fail which quality checks. Some of our samples failed 
quite a few quality metrics used by FastQC. However, this does not mean
that our samples should be thrown out! It is common to have some 
quality metrics fail, which may or may not be a problem for your 
downstream application. For our workflow, we will remove some low-quality sequences to reduce our false-positive rate due to 
sequencing errors.</p>
<p>To accomplish this, we will use a program called
<a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a>. This 
useful tool filters poor quality reads and trims poor quality bases 
from the specified samples.</p>
<h2 id="trimmomatic-options">Trimmomatic options<a class="headerlink" href="#trimmomatic-options" title="Permanent link">#</a></h2>
<p>Trimmomatic has a variety of options to accomplish its task. 
If we run the following command, we can see some of its options:</p>
<p class="bash"><div class="highlight"><pre><span></span><code>$ trimmomatic
</code></pre></div></p>
<p class="output">Which will give you the following output:
<div class="highlight"><pre><span></span><code>Usage: 
       PE [-version] [-threads &lt;threads&gt;] [-phred33|-phred64] [-trimlog &lt;trimLogFile&gt;] [-summary &lt;statsSummaryFile&gt;] [-quiet] [-validatePairs] [-basein &lt;inputBase&gt; | &lt;inputFile1&gt; &lt;inputFile2&gt;] [-baseout &lt;outputBase&gt; | &lt;outputFile1P&gt; &lt;outputFile1U&gt; &lt;outputFile2P&gt; &lt;outputFile2U&gt;] &lt;trimmer1&gt;...
   or: 
       SE [-version] [-threads &lt;threads&gt;] [-phred33|-phred64] [-trimlog &lt;trimLogFile&gt;] [-summary &lt;statsSummaryFile&gt;] [-quiet] &lt;inputFile&gt; &lt;outputFile&gt; &lt;trimmer1&gt;...
   or: 
       -version
</code></pre></div></p>
<p>This output shows that we must first specify whether we have paired-end (<code>PE</code>) or single-end (<code>SE</code>) reads. Next, we will specify with which flags we 
want to run Trimmomatic. For example, you can specify <code>threads</code> 
to indicate the number of processors on your computer that you want Trimmomatic 
to use. In most cases, using multiple threads(processors) can help to run the 
trimming faster. These flags are unnecessary, but they can give you more control
over the command. The flags are followed by <strong>positional arguments</strong>, meaning 
the order in which you specify them is essential. In paired-end mode, Trimmomatic 
expects the two input files and then the names of the output files. These files are 
described below. While in single-end mode, Trimmomatic will expect one file 
as input, after which you can enter the optional settings and, lastly, the 
name of the output file.</p>
<table>
<thead>
<tr>
<th>Option</th>
<th>Meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td>\<inputFile1></td>
<td>input forward reads to be trimmed. Typically the file name will contain an <code>_1</code> or <code>_R1</code> in the name.</td>
</tr>
<tr>
<td>\<inputFile2></td>
<td>Input reverse reads to be trimmed. Typically the file name will contain an <code>_2</code> or <code>_R2</code> in the name.</td>
</tr>
<tr>
<td>\<outputFile1P></td>
<td>Output file that contains surviving pairs from the <code>_1</code> file.</td>
</tr>
<tr>
<td>\<outputFile1U></td>
<td>Output file that contains orphaned reads from the <code>_1</code> file.</td>
</tr>
<tr>
<td>\<outputFile2P></td>
<td>Output file that contains surviving pairs from the <code>_2</code> file.</td>
</tr>
<tr>
<td>\<outputFile2U></td>
<td>Output file that contains orphaned reads from the <code>_2</code> file.</td>
</tr>
</tbody>
</table>
<p>The last thing Trimmomatic expects to see is the trimming parameters:</p>
<table>
<thead>
<tr>
<th>step</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>ILLUMINACLIP</code></td>
<td>Perform adapter removal.</td>
</tr>
<tr>
<td><code>SLIDINGWINDOW</code></td>
<td>Perform sliding window trimming, cutting once the average quality within the window falls below a threshold.</td>
</tr>
<tr>
<td><code>LEADING</code></td>
<td>Cut bases off the start of a read if below a threshold quality.</td>
</tr>
<tr>
<td><code>TRAILING</code></td>
<td>Cut bases off the end of a read if below a threshold quality.</td>
</tr>
<tr>
<td><code>CROP</code></td>
<td>Cut the read to a specified length.</td>
</tr>
<tr>
<td><code>HEADCROP</code></td>
<td>Cut the specified number of bases from the start of the read.</td>
</tr>
<tr>
<td><code>MINLEN</code></td>
<td>Drop an entire read if it is below a specified length.</td>
</tr>
<tr>
<td><code>TOPHRED33</code></td>
<td>Convert quality scores to Phred-33.</td>
</tr>
<tr>
<td><code>TOPHRED64</code></td>
<td>Convert quality scores to Phred-64.</td>
</tr>
</tbody>
</table>
<p>Understanding the steps you are using to
clean your data is essential. We will use only a few options and trimming steps in our
analysis. For more information about the Trimmomatic arguments
and options, see <a href="http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf">the Trimmomatic manual</a>.</p>
<p><a href="../fig/03-03-01.png">
  <img src="../fig/03-03-01.png" alt="Diagram showing the parts of the sequence that are reviewed by each parameter and the parts that are maintained or discarded at the end of the process. The Illuminaclip parameter removes the adapters, and the SlidingWindow scans the read by sections and removes a part of the read below the quality threshold. We remain with a trimmed read that has a valid quality." />
</a></p>
<p>However, a complete command for Trimmomatic will look something like the command below. This command is an example and will not work, as we do not have the files it refers to:</p>
<p class="bash"><div class="highlight"><pre><span></span><code>$ trimmomatic PE -threads 4 SRR_1056_1.fastq SRR_1056_2.fastq \
              SRR_1056_1.trimmed.fastq SRR_1056_1un.trimmed.fastq \
              SRR_1056_2.trimmed.fastq SRR_1056_2un.trimmed.fastq \
              ILLUMINACLIP:SRR_adapters.fa SLIDINGWINDOW:4:20
</code></pre></div></p>
<p>In this example, we have told Trimmomatic:</p>
<table>
<thead>
<tr>
<th>code</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>PE</code></td>
<td>that it will be taking a paired-end file as input</td>
</tr>
<tr>
<td><code>-threads 4</code></td>
<td>to use four computing threads to run (this will speed up our run)</td>
</tr>
<tr>
<td><code>SRR_1056_1.fastq</code></td>
<td>the first input file name. Forward</td>
</tr>
<tr>
<td><code>SRR_1056_2.fastq</code></td>
<td>the second input file name. Reverse</td>
</tr>
<tr>
<td><code>SRR_1056_1.trimmed.fastq</code></td>
<td>the output file for surviving pairs from the <code>_1</code> file</td>
</tr>
<tr>
<td><code>SRR_1056_1un.trimmed.fastq</code></td>
<td>the output file for orphaned reads from the <code>_1</code> file</td>
</tr>
<tr>
<td><code>SRR_1056_2.trimmed.fastq</code></td>
<td>the output file for surviving pairs from the <code>_2</code> file</td>
</tr>
<tr>
<td><code>SRR_1056_2un.trimmed.fastq</code></td>
<td>the output file for orphaned reads from the <code>_2</code> file</td>
</tr>
<tr>
<td><code>ILLUMINACLIP:SRR_adapters.fa</code></td>
<td>to clip the Illumina adapters from the input file using the adapter sequences listed in <code>SRR_adapters.fa</code></td>
</tr>
<tr>
<td><code>SLIDINGWINDOW:4:20</code></td>
<td>to use a sliding window of size 4 that will remove bases if their Phred score is below 20</td>
</tr>
</tbody>
</table>
<h2 id="running-trimmomatic-on-galaxy">Running Trimmomatic on Galaxy<a class="headerlink" href="#running-trimmomatic-on-galaxy" title="Permanent link">#</a></h2>
<p>Now, we will run Trimmomatic on our data.
Instead of using a command, we are going to use Galaxy again !</p>
<p>We are going to run Trimmomatic on one of our paired-end samples.
First, we will do it with JP4D.</p>
<h3 id="create-a-new-history">Create a new history<a class="headerlink" href="#create-a-new-history" title="Permanent link">#</a></h3>
<p>First, let's create a new history, and name it "Trimming and Filtering"
(Ajouter une image de creation d'history).
Then using the History Multiview, we can move the fastq from the previous analysis (FastQC). (JP4D collection)
Then, we can type "Trimmomatic" inside the "Tools" .  </p>
<p>While using FastQC, we saw that Universal adapters were present 
in our samples. The adapter sequences came with the installation of 
Trimmomatic and it is located in our current directory in the database  <code>TruSeq3</code>.</p>
<p>We will also use a sliding window of size 4 that will remove bases if their
Phred score is below 20 (like in our example above). We will also
discard any reads that do not have at least 25 bases remaining after
this trimming step.
(Put a picture of the parameters to use inside Trimmomatic)</p>
<p>Once you have selected all the good parameters, you can click on "Run".
This command will take a few minutes to run.
Open the output " Trimmomatic on collection X (log file)</p>
<p class="output"><div class="highlight"><pre><span></span><code>TrimmomaticPE: Started with arguments:
 JP4D_R1.fastq.gz JP4D_R2.fastq.gz JP4D_R1.trim.fastq.gz JP4D_R1un.trim.fastq.gz JP4D_R2.trim.fastq.gz JP4D_R2un.trim.fastq.gz SLIDINGWINDOW:4:20 MINLEN:35 ILLUMINACLIP:TruSeq3-PE.fa:2:40:15
Multiple cores found: Using 2 threads
Using PrefixPair: &#39;TACACTCTTTCCCTACACGACGCTCTTCCGATCT&#39; and &#39;GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT&#39;
ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
Quality encoding detected as phred33
Input Read Pairs: 1123987 Both Surviving: 751427 (66.85%) Forward Only Surviving: 341434 (30.38%) Reverse Only Surviving: 11303 (1.01%) Dropped: 19823 (1.76%)
TrimmomaticPE: Completed successfully
</code></pre></div></p>
<blockquote>
<h2 id="exercise-1-what-did-trimmomatic-do">Exercise 1: What did Trimmomatic do?<a class="headerlink" href="#exercise-1-what-did-trimmomatic-do" title="Permanent link">#</a></h2>
<p>Use the output from your Trimmomatic command to answer the
following questions.</p>
<p>1) What percent of reads did we discard from our sample?<br />
2) What percent of reads did we keep both pairs?</p>
<blockquote>
<h2 id="solution">Solution<a class="headerlink" href="#solution" title="Permanent link">#</a></h2>
<p class="challenge">1) 1.76%  <br />
2) 66.85%
{: .solution}</p>
</blockquote>
</blockquote>
<p>You may have noticed that Trimmomatic automatically detected the
quality encoding of our sample (phred33). It is always a good idea to
double-check this or manually enter the quality encoding.</p>
<h2 id="exercise-2">Exercise 2<a class="headerlink" href="#exercise-2" title="Permanent link">#</a></h2>
<p>The output files are also FASTQ files. </p>
<div class="admonition question">
<p class="admonition-title">Should the output fastqfile file be smaller or bigger than the input file</p>
<pre><code>??? success "It should be smaller than our input file because we have removed reads."
</code></pre>
</div>
<p>We have just successfully run Trimmomatic on one of our FASTQ files!
Now we need to do it on our other sample : JC1A.</p>
<p>We have completed the trimming and filtering steps of our quality
control process! </p>
<blockquote>
<h2 id="bonus-exercise-quality-test-after-trimming">Bonus Exercise: Quality test after trimming<a class="headerlink" href="#bonus-exercise-quality-test-after-trimming" title="Permanent link">#</a></h2>
<p>Now that our samples have gone through quality control, they should perform
better on the quality tests run by FastQC. </p>
</blockquote>
<p>Rerun the FastQC Analysis to check if our sequences have been succesfully filtered and trimmed.</p>
<p>After trimming and filtering, our overall quality is much higher, 
  we have a distribution of sequence lengths, and more samples pass 
  adapter content. However, quality trimming is not perfect, and some
  programs are better at removing some sequences than others. Trimmomatic 
  did pretty well, though, and its performance is good enough for our workflow.</p>
<p>Now that the Quality Control Process step is done, we can move on to the Taxoxomic assignement of our samples.   </p>







  
  






                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../03-assessing-read-quality/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Assessing Read Quality">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Assessing Read Quality
              </div>
            </div>
          </a>
        
        
          
          <a href="../05-taxonomic/" class="md-footer__link md-footer__link--next" aria-label="Next: Taxonomy Profiling">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Taxonomy Profiling
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["announce.dismiss", "content.action.edit", "content.action.view", "content.code.annotate", "content.code.copy", "content.tooltips", "header.autohide", "navigation.footer", "navigation.indexes", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow", "toc.integrate"], "search": "../../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
    
  </body>
</html>